package.json

```diff
@@ -277,6 +277,24 @@
                 "command": "jupyter.runAndDebugCell",
                 "key": "ctrl+alt+shift+enter",
                 "mac": "ctrl+shift+enter"
+            },
+            {
+                "command": "jupyter.runByLine",
+                "key": "f10",
+                "mac": "f10",
+                "when": "!jupyter.notebookeditor.debuggingInProgress && !jupyter.notebookeditor.runByLineInProgress"
+            },
+            {
+                "command": "jupyter.runByLineContinue",
+                "key": "f10",
+                "mac": "f10",
+                "when": "jupyter.notebookeditor.runByLineInProgress"
```

Comment:
Out of interest, what happens if I'm debugging something in my workspace (not a notebook) and then I start run by line?

Reply:
The debugging manager checks for the active editor [here](https://github.com/microsoft/vscode-jupyter/blob/1dda6c08155a6ee2a3e240ddd5fcb6d1091a2a1c/src/client/debugger/jupyter/debuggingManager.ts#L217). Unless you're on the editor, nothing will happen on the run by line side.